app/main/views/audits.py

```diff
@@ -67,6 +95,47 @@ def list_audits():
     )
 
 
+@main.route('/audit-events', methods=['POST'])
+def create_audit_event():
+    json_payload = get_json_from_request()  # TODO test
+    json_has_required_keys(json_payload, ['auditEvents'])  # TODO test
+    audit_event_data = json_payload['auditEvents']
+    json_has_required_keys(audit_event_data, ["type", "user", "data"])
+
+    if 'objectType' not in audit_event_data:
+        if 'objectId' in audit_event_data:
+            abort(400, "object ID cannot be provided without an object type")
+        db_object = None
+    else:
+        if audit_event_data['objectType'] not in AUDIT_OBJECT_TYPES:
+            abort(400, "invalid object type supplied")
+        if 'objectId' not in audit_event_data:
+            abort(400, "object type cannot be provided without an object ID")
+        model = AUDIT_OBJECT_TYPES[audit_event_data['objectType']]
+        id_field = AUDIT_OBJECT_ID_FIELDS[audit_event_data['objectType']]
+        db_objects = model.query.filter(
+            id_field == audit_event_data['objectId']
+        ).all()
+        if len(db_objects) != 1:
+            abort(400, "referenced object does not exist")
```

Comment:
We had a chat about this and it boils down to a difference of opinion on what an audit event is (is it a message attached to a database object or is it a message that may or may not have a database object attached to it?). I'm going to merge this for now and add a note for us to figure this out.